src/routes/api/hotels.route.js

```diff
@@ -0,0 +1,155 @@
+import express from 'express';
+import hotels from '../../controllers/hotels.controller';
+import catchErrors from '../../utils/helper';
+import fileService from '../../services/files.service';
+import { validation } from '../../validation/validation';
+import { authorizeTravelAdmin } from '../../middlewares/roleAuthorization';
+import { verifyUser } from '../../middlewares/checkToken';
+
+
+const router = express.Router();
+
+/**
+ * @swagger
+ *
+ * /hotels:
+ *   post:
+ *     summary: Add new Hotel
+ *     description: This allows travel admins to add accomodation facility
+ *     tags:
+ *       - Accommodation
+ *     requestBody:
+ *       content:
+ *         multipart/form-data:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               name:
+ *                 type: integer
+ *               image:
+ *                 type: string
+ *                 format: binary
+ *               description:
+ *                 type: string
+ *               services:
+ *                 type: string
+ *               street:
+ *                 type: string
+ *               city:
+ *                 type: string
+ *               country:
+ *                 type: string
+ *     produces:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               status:
+ *                 type: string
+ *               message:
+ *                 type: string
+ *               data:
+ *                 type: object
+ *                 properties:
+ *                   id:
+ *                     type: integer
+ *                   locationId:
+ *                     type: string
+ *                   name:
+ *                     type: string
+ *                   image:
+ *                     type: string
+ *                   street:
+ *                     type: string
+ *                   description:
+ *                     type: string
+ *                   services:
+ *                     type: string
+ *                   updatedAt:
+ *                     type: string
+ *                   createAt:
+ *                     type: string
+ *     responses:
+ *       201:
+ *         description: success
+ */
+router.post(
+  '/hotels',
+  verifyUser,
+  authorizeTravelAdmin,
```

Comment:
Yeah, we can do that, but he will also be responsible for managing rooms for that hotel. because the user who creates a hotel becomes the owner of it and is the one who is responsible to add rooms.

Reply:
Here is how it should be done:
```
import Responses from '../utils/response';

const checkIsAuthorizedRole = (allowed) => (req, res, next) => {
  const { role } = res.locals.user;
  if (allowed.includes(role)) {
    next();
  } else {
    Responses.handleError(403, 'insufficient privileges', res);
  }
};

export { checkIsAuthorizedRole };
```
Where `allowed` parameter is an array of all allowed roles string names.
And the be called like this:

```
router.post(
  '/hotels',
  verifyUser,
  checkIsAuthorizedRole(['travel_administrator']),
  fileService.upload('image'),
  validation,
  catchErrors(hotels.createHotel)
);
```

Reply:
Thank you very much @gildniy let me use this.

Reply:
### Awesome, this worked thanks @gildniy